internal/photoprism/config.go

```diff
@@ -1,38 +1,45 @@
 package photoprism
 
 import (
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
 	"github.com/jinzhu/gorm"
-	_ "github.com/jinzhu/gorm/dialects/mssql"
+	_ "github.com/jinzhu/gorm/dialects/mssql" // Import gorm drivers
 	_ "github.com/jinzhu/gorm/dialects/mysql"
 	_ "github.com/jinzhu/gorm/dialects/postgres"
 	_ "github.com/jinzhu/gorm/dialects/sqlite"
-	"github.com/kylelemons/go-gypsy/yaml"
-	. "github.com/photoprism/photoprism/internal/models"
+	"github.com/photoprism/photoprism/internal/models"
 	"github.com/urfave/cli"
-	"log"
-	"os"
-	"time"
+	"gopkg.in/yaml.v2"
 )
 
+// Config provides a struct in which application configuration is stored.
 type Config struct {
-	Debug          bool
+	Debug          bool `yaml:"debug"`
 	ConfigFile     string
 	ServerIP       string
-	ServerPort     int
-	ServerMode     string
-	AssetsPath     string
-	ThumbnailsPath string
-	OriginalsPath  string
-	ImportPath     string
-	ExportPath     string
-	DarktableCli   string
-	DatabaseDriver string
-	DatabaseDsn    string
+	ServerPort     int    `yaml:"server-port"`
+	ServerMode     string `yaml:"server-mode"`
+	AssetsPath     string `yaml:"assets-path"`
+	ThumbnailsPath string `yaml:"thumbnails-path"`
```

Comment:
As discussed in #50, I think it's better to use functions in this particular case. I'll try to document that for others inline and/or in the developer guide. Thank you for the valuable feedback and suggestions!